spec/unit/facter/sensu_version_spec.rb

```diff
@@ -0,0 +1,45 @@
+require "spec_helper"
+
+describe "Facter::Util::Fact" do
+  context "on not windows" do
+    before {
+      Facter.clear
+      Facter.fact(:kernel).stubs(:value).returns('NotWindows')
+    }
+    describe 'returns sensu version when present' do
+        it do
+          File.stubs(:exists?).with('/opt/sensu/embedded/bin/sensu-client').returns(true)
+          Facter::Util::Resolution.expects(:exec).with('/opt/sensu/embedded/bin/sensu-client --version 2>&1').returns('0.23.3')
+          expect(Facter.value(:sensu_version)).to eql('0.23.3')
```

Comment:
seems these tests are failing

Reply:
You can use this as a reference for stubbing facts

https://github.com/ghoneycutt/puppet-module-ssh/blob/master/spec/unit/facter/ssh_spec.rb